File: jdroid-java-googleplay-publisher/src/test/java/com/jdroid/android/googleplay/publisher/AppContextTest.java
package com.jdroid.android.googleplay.publisher;

import org.junit.Assert;
import org.junit.Test;

public class AppContextTest {
	
	@Test
	public void testUserFraction() {
		AppContext appContext = new AppContext();
		appContext.setUserFraction(0.001);
		Assert.assertEquals(0.1, appContext.getUserPercentage(), 0);
		
		appContext = new AppContext();
		appContext.setUserFraction(0.01);
		Assert.assertEquals(1, appContext.getUserPercentage(), 0);
		
		appContext = new AppContext();
		appContext.setUserFraction(0.1);
		Assert.assertEquals(10, appContext.getUserPercentage(), 0);
		
		appContext = new AppContext();
		appContext.setUserFraction(0.5);
		Assert.assertEquals(50, appContext.getUserPercentage(), 0);
	}
	
	@Test
	public void testUserPercentage() {
		AppContext appContext = new AppContext();
		appContext.setUserPercentage(0.1);
		Assert.assertEquals(0.001, appContext.getUserFraction(), 0);
		
		appContext = new AppContext();
		appContext.setUserPercentage(1D);
		Assert.assertEquals(0.01, appContext.getUserFraction(), 0);
		
		appContext = new AppContext();
		appContext.setUserPercentage(10D);
		Assert.assertEquals(0.1, appContext.getUserFraction(), 0);
		
		appContext = new AppContext();
		appContext.setUserPercentage(50D);
		Assert.assertEquals(0.5, appContext.getUserFraction(), 0);
	}
}

File: google-play-api-kotlin/src/test/java/com/dipien/google/play/api/AppContextTest.kt
package com.dipien.google.play.api

import org.junit.Assert
import org.junit.Test

class AppContextTest {
    @Test
    fun testUserFraction() {
        var appContext = AppContext()
        appContext.userFraction = 0.001
        Assert.assertEquals(0.1, appContext.userPercentage!!, 0.0)
        appContext = AppContext()
        appContext.userFraction = 0.01
        Assert.assertEquals(1.0, appContext.userPercentage!!, 0.0)
        appContext = AppContext()
        appContext.userFraction = 0.1
        Assert.assertEquals(10.0, appContext.userPercentage!!, 0.0)
        appContext = AppContext()
        appContext.userFraction = 0.5
        Assert.assertEquals(50.0, appContext.userPercentage!!, 0.0)
    }

    @Test
    fun testUserPercentage() {
        var appContext = AppContext()
        appContext.userPercentage = 0.1
        Assert.assertEquals(0.001, appContext.userFraction!!, 0.0)
        appContext = AppContext()
        appContext.userPercentage = 1.0
        Assert.assertEquals(0.01, appContext.userFraction!!, 0.0)
        appContext = AppContext()
        appContext.userPercentage = 10.0
        Assert.assertEquals(0.1, appContext.userFraction!!, 0.0)
        appContext = AppContext()
        appContext.userPercentage = 50.0
        Assert.assertEquals(0.5, appContext.userFraction!!, 0.0)
    }
}